tests/unit/amo/api/test_collections.js

```diff
@@ -0,0 +1,159 @@
+import * as api from 'core/api';
+import {
+  getCollectionAddons,
+  getCollectionDetail,
+} from 'amo/api/collections';
+import { parsePage } from 'core/utils';
+import { createApiResponse } from 'tests/unit/helpers';
+import {
+  dispatchClientMetadata,
+  dispatchSignInActions,
+} from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  let mockApi;
+
+  const getParams = ({
+    apiState = dispatchClientMetadata().store.getState().api,
+    ...otherParams
+  } = {}) => {
+    return {
+      api: apiState,
+      slug: 'some-slug',
+      user: 'user-id-or-name',
+      ...otherParams,
+    };
+  };
+
+  beforeEach(() => {
+    mockApi = sinon.mock(api);
+  });
+
+  describe('getCollectionDetail', () => {
+    it('throws an error when slug is missing', () => {
+      const params = getParams();
+      delete params.slug;
+
+      expect(() => {
+        getCollectionDetail(params);
+      }).toThrow('slug is required');
+    });
+
+    it('throws an error when user is missing', () => {
+      const params = getParams();
+      delete params.user;
+
+      expect(() => {
+        getCollectionDetail(params);
+      }).toThrow('user is required');
+    });
+
+    it('calls the collection detail API', () => {
+      const apiState = dispatchClientMetadata().store.getState().api;
+      const params = getParams({ api: apiState });
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          auth: false,
+          endpoint: 'accounts/account/user-id-or-name/collections/some-slug',
+          state: apiState,
+        })
+        .once()
+        .returns(createApiResponse());
+
+      return getCollectionDetail(params)
+        .then(() => {
+          mockApi.verify();
+        });
+    });
+
+    it('makes an authenticated request when API state allows it', () => {
+      const apiState = dispatchSignInActions().store.getState().api;
+      const params = getParams({ apiState });
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          auth: true,
+          endpoint:
+          'accounts/account/user-id-or-name/collections/some-slug',
```

Comment:
I think it helps for readability to indent continuations like this but we don't have a lint rule so meh

Reply:
Yes, and I think I edited the file and reformatted it entirely. Unfortunately, my vim does not want to keep such indentations 😢 

I'll fix this!